cli/tests/unit/streams_test.ts

```diff
@@ -468,3 +468,47 @@ Deno.test(async function readableStreamWithAggressiveResourceClose() {
   }
   assertEquals(await reasonPromise, "resource closed");
 });
+
+Deno.test(async function compressionStreamWritableMayBeAborted() {
+  await Promise.all([
+    new CompressionStream("gzip").writable.getWriter().abort(),
+    new CompressionStream("deflate").writable.getWriter().abort(),
+    new CompressionStream("deflate-raw").writable.getWriter().abort(),
+  ]);
+});
+
+Deno.test(async function compressionStreamReadableMayBeCancelled() {
+  await Promise.all([
+    new CompressionStream("gzip").readable.getReader().cancel(),
+    new CompressionStream("deflate").readable.getReader().cancel(),
+    new CompressionStream("deflate-raw").readable.getReader().cancel(),
+  ]);
+});
+
+Deno.test(async function decompressionStreamWritableMayBeAborted() {
+  await Promise.all([
+    assertRejects(
+      async () => {
+        await new DecompressionStream("gzip").writable.getWriter().abort();
+      },
+      TypeError,
+      "corrupt gzip stream does not have a matching checksum",
```

Comment:
I don't think we should throw on cancellation. This probably means we need a separate Rust op (or a second parameter) to just throw away the results.

Reply:
I actually started work on a separate Rust op, but then decided that we should probably give the user the chance to handle this case as they see fit. Thinking of intentional abortion on the user side where the user wants to continue work with the results received so far, it would be good to let the user know if they need to perform some manual cleanup on the last bytes received or can use the slice as is. This being said, I don't know the specifics of gzip, deflate and deflate-raw enough to say if a decompressed slice is even usable, this being implemented as a stream I suppose it is. If it is, I'd at least give the user the option to request error bubbling, maybe with an options object to the `CompressionStream` and `DecompressionStream` constructors.

Reply:
I'm not entirely sure here...  I might have to hand this one off to someone with a better intuition. 

Cancellation on a TextDecoderStream in Firefox appears to reject the read with the cancellation rather than throwing from the writer abort:

```
(async () => { let td = new TextDecoderStream(); td.writable.abort("hmm"); await td.readable.getReader().read() })()
Promise { <state>: "rejected", <reason>: "hmm" }
```

I wonder if this is the right way to do things here.

Reply:
Yeah, we should follow browser behaviour here (`.cancel()` should not throw)

`cancel()` should just `close()` the underlying resource I think.

Reply:
Agreed, I pushed the necessary changes.